lib/home/widgets/collections_screen.dart

```diff
@@ -12,77 +13,82 @@ class CollectionsScreen extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return BlocBuilder<HomeBloc, HomeState>(
-      builder: (context, state) {
-        return Scaffold(
-            appBar: AppBar(title: const Text('Collections')),
-            body: ListView.builder(
-                padding: const EdgeInsets.all(2.0),
-                itemCount: state.gibsonsForms.length,
-                itemBuilder: (context, index) {
-                  // TODO: Refactor to a standalone widget
-                  return Slidable(
-                    key: Key(state.gibsonsForms[index]!.id),
-                    endActionPane: ActionPane(
-                      motion: const ScrollMotion(),
-                      children: [
-                        SlidableAction(
-                          onPressed: (context) => showDialog<String>(
+    return BlocBuilder<LoginBloc, LoginState>(builder: (context, loginState) {
+      return BlocBuilder<HomeBloc, HomeState>(
+        builder: (context, homeState) {
```

Comment:
Hmm, so far in the app we didn't have to nest BlocBuilders — just wondering, is there some other way of doing this? We might load the employee id from the shared prefs in Collection bloc, but I'm not sure if that would be better. Are there any other options to consider @sshazril ?

Reply:
Or we could save the employee id in the state of Collection & Recipe bloc when it is changed, but again, not sure if that is better.

Reply:
Yeah, the only place we presently nest BlocBuilders is for the data export page, which I suppose was only ever a temporary implementation. Having looked into it - there are [no plans](https://github.com/felangel/bloc/issues/1724) to add a MultiBlocProvider although the nesting implementation does work. That said, for this specific purpose, I think using a BlocListener for LoginBloc and a BlocBuilder for HomeBloc is the correct way forward. For your export, I believe [MultiBlocListener](https://pub.dev/documentation/flutter_bloc/latest/flutter_bloc/MultiBlocListener-class.html) is what would be the intended implementation.

Reply:
Am happy to leave this nesting as is for now and add a TODO to switch to BlocListener - just in case it isn't as simple as I envision...

Reply:
I understand, thanks for the review — seems like we should go for nested bloc builders for now!